build.rs

```diff
@@ -370,13 +370,28 @@ fn pregenerate_asm_main() {
         perlasm(&perlasm_src_dsts, asm_target);
 
         if asm_target.preassemble {
+            // Preassembly is currently only done for Windows targets.
+            assert_eq!(&asm_target.oss, &[WINDOWS]);
+            let os = WINDOWS;
+            let env = "msvc";
```

Comment:
Is it not done for Windows *-gnu targets though? 🤔

Reply:
@Alovchin91 Another excellent question! I submitted PR #1348 to make it clearer. I will then rebase this on top of #1348 if that makes sense. Then there won't be any "msvc" here and inspecting the rest of the code will make it clear that the preassembled object files are used by all windows targets. See this existing code in build.rs that determines whether to use the preassembled code:

```
if use_pregenerated && target.os == WINDOWS {
```

Reply:
Also, at least to me it makes sense to use the same preassembled object files for -gnu targets since ultimately the ABI is the same, at least for code that doesn't use the C standard library, which the assembly code doesn't.

Reply:
😅  Much clearer now indeed, thanks!

> Also, at least to me it makes sense to use the same preassembled object files for -gnu targets since ultimately the ABI is the same, at least for code that doesn't use the C standard library, which the assembly code doesn't.

I'm not super-professional here so the only thing I know is that GCC object files are usually not advised to be mixed with MSVC, but yet again — I'm dumb in this kind of questions 😬